src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/resource/ReportDescriptionAddendumResource.kt

```diff
@@ -0,0 +1,265 @@
+package uk.gov.justice.digital.hmpps.incidentreporting.resource
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import jakarta.validation.Valid
+import org.springframework.http.HttpStatus
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PatchMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.incidentreporting.SYSTEM_USERNAME
+import uk.gov.justice.digital.hmpps.incidentreporting.dto.DescriptionAddendum
+import uk.gov.justice.digital.hmpps.incidentreporting.dto.request.AddDescriptionAddendum
+import uk.gov.justice.digital.hmpps.incidentreporting.dto.request.UpdateDescriptionAddendum
+import uk.gov.justice.digital.hmpps.incidentreporting.service.WhatChanged
+import java.time.LocalDateTime
+import java.util.UUID
+
+@RestController
+@Validated
+class ReportDescriptionAddendumResource :
+  ReportRelatedObjectsResource<DescriptionAddendum, AddDescriptionAddendum, UpdateDescriptionAddendum>() {
+  @GetMapping("/description-addendums")
+  @ResponseStatus(HttpStatus.OK)
+  @PreAuthorize("hasRole('ROLE_VIEW_INCIDENT_REPORTS')")
+  @Operation(
+    summary = "Returns description addendums for this incident report",
+    description = "Requires role VIEW_INCIDENT_REPORTS",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Returns description addendums",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Missing required role. Requires the VIEW_INCIDENT_REPORTS role",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Data not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  @Transactional(readOnly = true)
```

Comment:
I notice this pattern in the service with having the transaction in the resource, this isn't ideal, and also non-standard from other APIs.  The transaction boundary should be at the service level.  Its not a problem yet as we don't have any listening clients to events but when we do the data will potentially not yet be committed when the event fires.  Also if the transaction fails the event will be emitted and the data never updated